rollout/trafficrouting.go

```diff
@@ -237,10 +237,7 @@ func (c *rolloutContext) reconcileTrafficRouting() error {
 				desiredWeight = 100
 			}
 		}
-
-		// We need to check for Generation > 1 because when we first install the rollout we run step 0 this prevents that.
-		// We could also probably use c.newRS == nil || c.newRS.Status.AvailableReplicas == 0
-		if currentStep != nil && c.rollout.ObjectMeta.Generation > 1 {
```

Comment:
I don't think this is the correct thing to do here these steps should not run when a rollout is first installed into a cluster. What is your reasoning for this change?

Reply:
[while i](https://github.com/argoproj/argo-rollouts/issues/2780)  
When I followed the steps of this issue to run rollback, I found that VitrualService was not set correctly.  
Only when the generation of rollout is greater than 1 will VirtualService be correctly changed

Reply:
I think this is expected behavior though, when a rollout is first installed we do not run any of the steps because there is nothing to canary too because the rollouts is being installed for the very first time, any subsequent updates should will bump generation so that we run the `steps`

Reply:
But when I update the image of deployment  referenced by `WorkloadRef`, the generation of Rollout does not change. I think this is because we didn't update the Rollout resource directly. 

Maybe we should use the condition `c.newRS == nil || c.newRS.Status.AvailableReplicas == 0` as the comment indicates?

Reply:
ahh sure, this is with workloadRef that makes more sense. Using the other condition makes sense should also add test case for workloadRef as well

Reply:
I modify it use `currentStep instead= Nil&&(c.newRS!=nil&&c.newRS. Status. AvailableReplicas=c.newRS.Status. Replicas)` made a judgment and added some unit tests

Reply:
@zachaller please take a look, communicate with me if there are any problems, thanks

Reply:
@alexmt please take a look, thanks